src/main/java/net/wurstclient/settings/RoundingPrecisionSetting.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2014-2023 Wurst-Imperium and contributors.
+ *
+ * This source code is subject to the terms of the GNU General Public
+ * License, version 3. If a copy of the GPL was not distributed with this
+ * file, You can obtain one at: https://www.gnu.org/licenses/gpl-3.0.txt
+ */
+package net.wurstclient.settings;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+
+public final class RoundingPrecisionSetting extends SliderSetting
+{
+	private static final DecimalFormatSymbols SYMBOLS =
+		new DecimalFormatSymbols(Locale.ENGLISH);
+	
+	private final DecimalFormat[] FORMATS;
+	
+	public RoundingPrecisionSetting(String name, String description, int value,
+		int min, int max)
+	{
+		super(name, description, value, min, max, 1,
+			ValueDisplay.ROUNDING_PRECISION);
+		
+		FORMATS = new DecimalFormat[max + 1];
+	}
+	
+	public DecimalFormat getFormat()
+	{
+		int value = getValueI();
+		
+		if(FORMATS[value] == null)
+		{
+			String pattern = "0";
+			if(value > 0)
+				pattern += "." + "#".repeat(value);
+			
+			FORMATS[value] = new DecimalFormat(pattern, SYMBOLS);
+		}
+		
+		return FORMATS[value];
+	}
+	
+	public String format(double value)
+	{
+		return getFormat().format(value);
+	}

```

Comment:
`getValueI()` must return a value no lower than `min` and no higher than `max`.

Since the length of the `FORMATS` array is `max + 1`, that means we don't have to worry about `getValueI()` returning a value that is too high. However, `getValueI()` could return a value that is too low if `min` is set to a value lower than `0`.

I will fix this by adding a check in the constructor that disallows negative `min` values:
```java
if(min < 0)
	throw new IllegalArgumentException(
		"min must be greater than or equal to 0");
```